File: CS/SchedulingDemo.Wpf/ViewModels/AppointmentCustomizationDemoViewModel.cs
using System.Collections.Generic;
using DevExpress.Mvvm.POCO;

namespace SchedulingDemo.ViewModels {
    public class AppointmentCustomizationDemoViewModel {
        readonly CarsData data = new CarsData();
        protected AppointmentCustomizationDemoViewModel() {
            CustomFlyout = false;
            CustomTemplate = CustomTemplate_AllDay = false;
            ShowStatus = ShowStatus_AllDay = true;
            ShowLocation = ShowLocation_AllDay = true;
            ShowDescription = ShowDescription_AllDay = true;
            ShowInterval = ShowInterval_AllDay = true;
            IntervalStringFormat = IntervalStringFormat_AllDay = "{0:hh:mm tt} - {1:hh:mm tt}";
        }
        public virtual List<CarScheduling> Events { get { return this.data.Events; } }
        public virtual List<Car> Cars { get { return this.data.Cars; } }

        public virtual bool CustomFlyout { get; set; }

        public virtual bool CustomTemplate { get; set; }
        public virtual bool ShowStatus { get; set; }
        public virtual bool ShowLocation { get; set; }
        public virtual bool ShowDescription { get; set; }
        public virtual bool ShowInterval { get; set; }
        public virtual string IntervalStringFormat { get; set; }

        public virtual bool CustomTemplate_AllDay { get; set; }
        public virtual bool ShowStatus_AllDay { get; set; }
        public virtual bool ShowLocation_AllDay { get; set; }
        public virtual bool ShowDescription_AllDay { get; set; }
        public virtual bool ShowInterval_AllDay { get; set; }
        public virtual string IntervalStringFormat_AllDay { get; set; }
    }
}

File: VB/SchedulingDemo.Wpf/ViewModels/AppointmentCustomizationDemoViewModel.vb
Imports System.Collections.Generic
Imports DevExpress.Mvvm.POCO

Namespace SchedulingDemo.ViewModels
    Public Class AppointmentCustomizationDemoViewModel
        Private ReadOnly data As New CarsData()
        Protected Sub New()
            CustomFlyout = False
            CustomTemplate_AllDay = False
            CustomTemplate = CustomTemplate_AllDay
            ShowStatus_AllDay = True
            ShowStatus = ShowStatus_AllDay
            ShowLocation_AllDay = True
            ShowLocation = ShowLocation_AllDay
            ShowDescription_AllDay = True
            ShowDescription = ShowDescription_AllDay
            ShowInterval_AllDay = True
            ShowInterval = ShowInterval_AllDay
            IntervalStringFormat_AllDay = "{0:hh:mm tt} - {1:hh:mm tt}"
            IntervalStringFormat = IntervalStringFormat_AllDay
        End Sub
        Public Overridable ReadOnly Property Events() As List(Of CarScheduling)
            Get
                Return Me.data.Events
            End Get
        End Property
        Public Overridable ReadOnly Property Cars() As List(Of Car)
            Get
                Return Me.data.Cars
            End Get
        End Property

        Public Overridable Property CustomFlyout() As Boolean

        Public Overridable Property CustomTemplate() As Boolean
        Public Overridable Property ShowStatus() As Boolean
        Public Overridable Property ShowLocation() As Boolean
        Public Overridable Property ShowDescription() As Boolean
        Public Overridable Property ShowInterval() As Boolean
        Public Overridable Property IntervalStringFormat() As String

        Public Overridable Property CustomTemplate_AllDay() As Boolean
        Public Overridable Property ShowStatus_AllDay() As Boolean
        Public Overridable Property ShowLocation_AllDay() As Boolean
        Public Overridable Property ShowDescription_AllDay() As Boolean
        Public Overridable Property ShowInterval_AllDay() As Boolean
        Public Overridable Property IntervalStringFormat_AllDay() As String
    End Class
End Namespace
